bluesky/plan_stubs.py

```diff
@@ -295,6 +302,82 @@ def inner_mvr():
 movr = mvr  # synonym
 
 
+def rd(obj, *, default_value=0):
+    """Reads a single-value non-triggered object
+
+    This is a helper plan to get the scalar value out of a Device (such as an EpicsMotor or a
+    single EpicsSignal).
+
+    For devices that have more than one read key the following rules are used:
+
+    - if exactly 1 field is hinted that value is used
+    - if no fields are hinted and there is exactly 1 value in the
+      reading that value is used
+    - if more than one field is hinted an Exception is raised
+    - if no fields are hinted and there is more than one key in the reading an
+      Exception is raised
+
+    The devices is not triggered and this plan does not create any Events
+
+    Parameters
+    ----------
+    obj : Device
+        The device to be read
+
+    default_value : Any
+        The value to return when not running in a "live" RunEngine (when
+        ret = yield from obj.read() returns None)
+
+    Returns
+    -------
+    val : Any or None
+        The "single" value of the device
+
+    """
+    hints = obj.hints.get("fields", [])
```

Comment:
We normally access hints via describe(). I think this might be the first time we consult the attribute directly. It is definitely cheaper, so I think there is a case for taking this approach and effectively adding it as an optional attribute in the ophyd—bluesky interface.

It is strictly optional though—I think we have tests that ensure this—so this should be wrapped.
```suggestion
    hints = getattr(obj, 'hints', {}).get("fields", [])
```